tests/builder/app_settings.rs

```diff
@@ -263,6 +263,41 @@ fn infer_subcommands_pass_conflicting_aliases() {
     assert_eq!(m.subcommand_name(), Some("test"));
 }
 
+#[test]
+fn infer_long_flag_pass_conflicting_aliases() {
+    let m = Command::new("prog")
+        .infer_subcommands(true)
+        .subcommand(Command::new("test").long_flag_aliases(["testa", "t", "testb"]))
```

Comment:
Doesn't this also need a long flag that isn't an alias?

Reply:
Fixed in the parser to use aliases even if long_flag isn't specified. If that's not desirable, the other sensible behavior is to crash with an error.

Reply:
Thinking about this a bit more, why is there a distinction between `long_flag` and `long_flag_aliases`? Could we just combine those into one `long_flags` API?

Reply:
Maybe.  If you want to pursue that, I'd recommend creating a Discussion or Issue as  that would be out of scope of this PR.

Reply:
Ok, restored the old behavior and filed https://github.com/clap-rs/clap/issues/5138